internal/cmd/docker/push.go

```diff
@@ -0,0 +1,98 @@
+package docker
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/registry"
+	"github.com/docker/docker/client"
+	cmds "github.com/saucelabs/saucectl/internal/cmd"
+	"github.com/saucelabs/saucectl/internal/segment"
+	"github.com/saucelabs/saucectl/internal/usage"
+	"github.com/spf13/cobra"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+)
+
+func PushCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:          "push",
+		Short:        "Push a Docker image to the Sauce Labs Container Registry.",
+		SilenceUsage: true,
+		Args: func(cmd *cobra.Command, args []string) error {
```

Comment:
(optional) Since this command is only used for pushing to a Sauce Labs container registry, how hard do you think would it be to verify that the user specified the correct registry? Any decent regex we can use? If not, that's fine, but if we could, it might help avoid unnecessary support tickets because someone didn't read and comprehend the command description.